File: print_ll.c
#include <stdio.h>
#include <stdlib.h>
struct Node {
    int data;
    struct Node* next;
};
struct Node* push(struct Node* head, int data) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = data;
    newNode->next = head;
    return newNode;
}
void printLinkedList(struct Node* head) {
    struct Node* temp = head;
    while (temp != NULL) {
        printf("%d->", temp->data);
        temp = temp->next;
    }
    printf("NULL\n");
}
int main() {
    struct Node* head = NULL;
    head = push(head, 3);
    head = push(head, 2);
    head = push(head, 1);
    printLinkedList(head);
    return 0;
}

File: print_ll.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
def push(head, data):
    newNode = Node(data)
    newNode.next = head
    return newNode
def printLinkedList(head):
    temp = head
    while temp:
        print(temp.data)
        temp = temp.next
if __name__ == "__main__":
    head = None
    head = push(head, 3)
    head = push(head, 2)
    head = push(head, 1)
    printLinkedList(head)
